internal/repositories/account/account.go

```diff
@@ -0,0 +1,264 @@
+	all, err := models.KernelAccounts(models.KernelAccountWhere.OwnerAddress.EQ(signer.Bytes())).All(ctx, r.PDB.DBS().Reader)
+	if err != nil {
+		return nil
+	}
+
+	apiAccount := &gmodel.Account{
+		Signer: common.BytesToAddress(signer.Bytes()),
+	}
+
+	for _, acct := range all {
+		apiAccount.Kernel = append(apiAccount.Kernel, &gmodel.Kernel{
+			Address:   common.BytesToAddress(acct.Kernel),
+			CreatedAt: acct.CreatedAt,
+			Signer: &gmodel.Signer{
+				Address:     common.BytesToAddress(signer.Bytes()),
+				SignerAdded: acct.SignerAdded,
+			},
+		})
+	}
+
+	return apiAccount
+}
+
+func (r *Repository) GetAccountByKernel(ctx context.Context, kernel common.Address) *gmodel.Account {
+	all, err := models.KernelAccounts(models.KernelAccountWhere.Kernel.EQ(kernel.Bytes())).All(ctx, r.PDB.DBS().Reader)
+	if err != nil {
+		return nil
+	}
+
+	var apiAccount gmodel.Account
+	for idx, acct := range all {
+		if idx == 0 {
+			apiAccount.Signer = common.BytesToAddress(acct.OwnerAddress)
+		}
+		apiAccount.Kernel = append(apiAccount.Kernel, &gmodel.Kernel{
+			Address:   common.BytesToAddress(acct.Kernel),
+			CreatedAt: acct.CreatedAt,
+			Signer: &gmodel.Signer{
+				Address:     common.BytesToAddress(kernel.Bytes()),
+				SignerAdded: acct.SignerAdded,
+			},
+		})
+	}
+
+	return &apiAccount
+}
+
+func (r *Repository) GetAccountAddress(ctx context.Context, address common.Address) *gmodel.Account {
+	all, err := models.KernelAccounts(
+		qm.Expr(
+			qm.Or2(models.KernelAccountWhere.Kernel.EQ(address.Bytes())),
+			qm.Or2(models.KernelAccountWhere.OwnerAddress.EQ(address.Bytes())),
+		),
+	).All(ctx, r.PDB.DBS().Reader)
+	if err != nil {
+		return nil
+	}
+
+	var apiAccount gmodel.Account
+	for idx, acct := range all {
+		if idx == 0 {
+			apiAccount.Signer = common.BytesToAddress(acct.OwnerAddress)
+		}
+		apiAccount.Kernel = append(apiAccount.Kernel, &gmodel.Kernel{
+			Address:   common.BytesToAddress(acct.Kernel),
+			CreatedAt: acct.CreatedAt,
+			Signer: &gmodel.Signer{
+				Address:     common.BytesToAddress(acct.OwnerAddress),
+				SignerAdded: acct.SignerAdded,
+			},
+		})
+	}
+
+	return &apiAccount
+}
+
+func (r *Repository) GetAccounts(ctx context.Context, first *int, after *string, last *int, before *string, filterBy *gmodel.AccountFilter) (*gmodel.AccountConnection, error) {
+	limit, err := helpers.ValidateFirstLast(first, last, 30)
+	if err != nil {
+		return nil, err
+	}
+
+	queryMods := []qm.QueryMod{}
+	if filterBy != nil && filterBy.Signer != nil {
+		queryMods = append(queryMods, models.KernelAccountWhere.OwnerAddress.EQ(filterBy.Signer.Bytes()))
+	}
+
+	accountCount, err := models.KernelAccounts(queryMods...).Count(ctx, r.PDB.DBS().Reader)
+	if err != nil {
+		return nil, err
+	}
+
+	orderBy := " DESC"
+	if last != nil {
+		orderBy = " ASC"
+	}
+
+	queryMods = append(queryMods,
+		qm.Limit(limit+1),
+		qm.OrderBy(models.KernelAccountColumns.OwnerAddress+orderBy+", "+models.KernelAccountColumns.Kernel+orderBy),
```

Comment:
This is fine just showing you an alternative syntax
```suggestion
		qm.OrderBy(models.KernelAccountColumns.OwnerAddress+orderBy),
		qm.OrderBy(models.KernelAccountColumns.Kernel+orderBy),
```